Actions/ReadSecrets/ReadSecretsHelper.psm1

```diff
@@ -102,7 +102,7 @@ function ConnectAzureKeyVaultIfNeeded {
             Clear-AzContext -Scope Process
             Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
             Connect-AzAccount -ServicePrincipal -Tenant $tenantId -Credential $credential | Out-Null
-            Set-AzContext -Subscription $subscriptionId -Tenant $tenantId | Out-Null
+            Set-AzContext -SubscriptionId $subscriptionId | Out-Null
```

Comment:
| Out-Null 
what happens if this line fails ? dont we want to terminated the workflow in that case ?

Reply:
Out-Null doesn't stop things from failing - it is basically the same as $dummy = Set-AzContext

Reply:
the point is that we should fail instead of doing nothing.